crates/proto/src/xfer/dns_multiplexer.rs

```diff
@@ -300,7 +300,7 @@ where
         // store a Timeout for this message before sending
         let timeout = S::Time::delay_for(self.timeout_duration);
 
-        let (complete, receiver) = oneshot::channel();
+        let (complete, receiver) = mpsc::channel(CHANNEL_BUFFER_SIZE);
```

Comment:
At some point we'll want to figure out all the channels being created and figure out how to make the sizes configurable (this is just a comment, no changes required)...